code/modules/trader/crates/misc_variety.dm

```diff
@@ -0,0 +1,390 @@
+	name = "rootspeak nanodictionary"
+	desc += " Speaking like a tree is harder than you'd think."
+	tongue = all_languages[LANGUAGE_ROOTSPEAK]
+	progress_goal = 4
+	progress_time = 14 SECONDS
+	progress_fail_chance = 55
+
+/obj/item/dictionary/monkey/New()
+	name = "monkey nanodictionary"
+	desc += " For those with primal aspirations."
+	tongue = all_languages[LANGUAGE_MONKEY]
+	progress_goal = 20
+	progress_time = 2 SECONDS
+	progress_fail_chance = 10
+
+/obj/item/dictionary/tradeband/New()
+	name = "tradeband nanodictionary"
+	desc += " For the guild!"
+	tongue = all_languages[LANGUAGE_TRADEBAND]
+	progress_goal = 7
+	progress_time = 7 SECONDS
+	progress_fail_chance = 7
+
+/obj/item/dictionary/gutter/New()
+	name = "gutter nanodictionary"
+	desc += " Surprisingly easy to pick up!"
+	tongue = all_languages[LANGUAGE_GUTTER]
+	progress_goal = 4
+	progress_time = 8 SECONDS
+	progress_fail_chance = 5
+
+/obj/item/dictionary/mouse/New()
+	name = "custodial nanodictionary"
+	desc += " Can this one actually work? It's dripping..."
+	tongue = all_languages[LANGUAGE_MOUSE]
+	progress_goal = 20
+	progress_time = 15 SECONDS
+	progress_fail_chance = 20
+
+/obj/item/dictionary/clatter/New()
+	name = "clatter nanodictionary"
+	desc += " Unusually spooky."
+	tongue = all_languages[LANGUAGE_CLATTER]
+	progress_goal = 6
+	progress_time = 11 SECONDS
+	progress_fail_chance = 0
+
+/obj/item/dictionary/golem/New()
+	name = "golem nanodictionary"
+	desc += " What a grind."
+	tongue = all_languages[LANGUAGE_GOLEM]
+	progress_goal = 4
+	progress_time = 3 SECONDS
+	progress_fail_chance = 95
+
+/obj/item/dictionary/slime/New()
+	name = "slime nanodictionary"
+	desc += " Each word flows into the next."
+	tongue = all_languages[LANGUAGE_SLIME]
+	progress_goal = 6
+	progress_time = 5 SECONDS
+	progress_fail_chance = 33
+
+/obj/item/dictionary/grey/New()
+	name = "grey nanodictionary"
+	desc += " This language has no words for disobedience."
+	tongue = all_languages[LANGUAGE_GREY]
+	progress_goal = 6
+	progress_time = 8 SECONDS
+	progress_fail_chance = 0
+
+/obj/item/dictionary/martian/New()
+	name = "martian nanodictionary"
+	desc += " You need to supply the extra arms."
+	tongue = all_languages[LANGUAGE_MARTIAN]
+	progress_goal = 6
+	progress_time = 12 SECONDS
+	progress_fail_chance = 0
+
+/obj/item/dictionary/cult/New()
+	name = "Commentaries on the Arcane Tome"
+	desc = "A particularly dubious-looking nanodictionary with the stated goal of translating the Arcane Tome of the Blood Cult dedicated to Nar-Sie. According to the unknown author, 'each word is razor-fed and secret, thinner than cataclysms, tarnished like red-drink.'"
+	tongue = all_languages[LANGUAGE_CULT]
+	progress_goal = 4
+	progress_time = 60 SECONDS
+	progress_fail_chance = 60
+
+/obj/item/dictionary/xeno/New()
+	name = "unknown nanodictionary"
+	desc += " This one contains a lot of terrifying shrieks. Is this a language?"
+	tongue = all_languages[LANGUAGE_XENO]
+	progress_goal = 40
```

Comment:
40 seems excessive. 20 sounds more decent.

Reply:
Cult on average takes around 10 minutes to learn (10 uses failing 6 times succeeds 4 times)
Xeno takes on average half that time. Since it only costs 6 seconds to attempt to learn about it, with a fail chance of about 15%, you can get around 8 points of progress per minute, meaning you can learn it in around 5 minutes.

As long as you have a few minutes to sit around and work on it while you aren't moving about (something librarians are good at) you can easily get through learning a language quickly, but it's quite hard if you want to constantly move around.